java/vector/src/main/java/org/apache/arrow/vector/dictionary/DictionaryHashTable.java

```diff
@@ -135,7 +138,7 @@ public int getIndex(int indexInArray, ValueVector toEncode) {
     for (DictionaryHashTable.Entry e = table[index]; e != null ; e = e.next) {
       if ((e.hash == hash)) {
         int dictIndex = e.index;
-        if (dictionary.equals(dictIndex, toEncode, indexInArray)) {
+        if (matcher.match(dictIndex, toEncode, indexInArray)) {
```

Comment:
I think it's better to just make a `RangeEqualsVistor` each time here, e.g. 
`if (dictionary.accept(new RangeEqualVisitor(toEncode, dictIndex, indexInArray, 1))`

Reply:
Create an instsnce each time seems less effecient compared to reuse(The same reason for reset methed)? is there a special reason?

Reply:
@pravindra Also I want to hear your perspective on this, thanks :)

Reply:
fixed.

Reply:
Thanks, this looks much better to me. I prefer simplified code and to prevent class bloat, unless it's on a hot path and optimization is crucial. Even then, I don't think creating a new `RangeEqualsVisitor` would cause any noticeable performance loss.